core/src/main/java/feign/codec/InterceptorDecoder.java

```diff
@@ -0,0 +1,26 @@
+package feign.codec;
+
+import feign.Response;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+public class InterceptorDecoder implements Decoder {
```

Comment:
ResponseMappingDecoder seems a good name for me, I can change it!
Also you're right about the final modifier to the properties and some javadoc, I'm on it ;)